hypothesis-python/src/hypothesis/internal/conjecture/datatree.py

```diff
@@ -41,12 +42,21 @@
     sign_aware_lte,
 )
 
+if TYPE_CHECKING:
+    from typing import TypeAlias
+
+    from hypothesis.vendor.pretty import RepresentationPrinter
+
+ChildrenCacheValueT: "TypeAlias" = tuple[
+    Generator[ChoiceT, None, None], list[ChoiceT], set[ChoiceT]
+]
+
 
 class PreviouslyUnseenBehaviour(HypothesisException):
     pass
 
 
-def inconsistent_generation():
+def inconsistent_generation() -> NoReturn:
     raise FlakyStrategyDefinition(
         "Inconsistent data generation! Data generation behaved differently "
```

Comment:
I'd be tempted to instead assign `_FLAKY_STRAT_MSG = "..."` and then `raise FlakyStrategyDefinition(_FLAKY_STRAT_MSG)` in each location where we currently call this helper - it's a bit more obvious what's happening, and one fewer frame to look through if it ends up surfaced to users (which does happen)